tests/test_wheels.py

```diff
@@ -0,0 +1,88 @@
+import io
+import textwrap
+
+import pytest
+
+from installer.wheels import SuperfulousRecordColumnsWarning, parse_record_file
+
+
+def test_parse_wheel_record_simple():
+    record_content = textwrap.dedent(
+        """\
+        file.py,sha256=AVTFPZpEKzuHr7OvQZmhaU3LvwKz06AJw8mT\\_pNh2yI,3144
+        distribution-1.0.dist-info/RECORD,,
+        """
+    )
+    records = list(parse_record_file(io.StringIO(record_content)))
+    assert len(records) == 2
+
+    r0 = records[0]
+    assert r0.path.as_posix() == "file.py"
+    assert r0.hash_.name == "sha256"
+    assert r0.hash_.value == "AVTFPZpEKzuHr7OvQZmhaU3LvwKz06AJw8mT\\_pNh2yI"
+    assert r0.size == 3144
+
+    r1 = records[1]
+    assert r1.path.as_posix() == "distribution-1.0.dist-info/RECORD"
+    assert r1.hash_ is None
+    assert r1.size is None
+
+
+def test_parse_wheel_record_drop_superfulous():
+    record_content = textwrap.dedent(
+        """\
+        file.py,sha256=AVTFPZpEKzuHr7OvQZmhaU3LvwKz06AJw8mT\\_pNh2yI,3144,
+        distribution-1.0.dist-info/RECORD,,
+        """
+    )
+    record_io = io.StringIO(record_content)
+
+    with pytest.warns(SuperfulousRecordColumnsWarning) as ws:
+        records = list(parse_record_file(record_io))
+
+    assert len(ws) == 1
+    assert ws[0].message.args[0] == "Dropping columns [3:] from row 0"
```

Comment:
I wouldn't test the exact message as I find that to typically be brittle. I also don't think the message should be part of the API. At best I would check that `"0"` is in the message if you want to guarantee the troublesome row is mentioned.